bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Modal.java

```diff
@@ -0,0 +1,104 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2019 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.models;
+
+import org.jetbrains.annotations.NotNull;
+import org.osgi.annotation.versioning.ConsumerType;
+
+import com.adobe.cq.export.json.ComponentExporter;
+
+/**
+ * Defines the {@code Modal} Sling Model used for the
+ * {@code /apps/core/wcm/components/modal} component. This component currently
+ * supports using "#{modalId}" in the url
+ *
+ * @since com.adobe.cq.wcm.core.components.models 12.8.0
+ */
+@ConsumerType
+public interface Modal extends ComponentExporter {
+
+	/**
+	 * Returns the modal id based on the hash generated from the component path
+	 * 
+	 * @return modelId
+	 * @since com.adobe.cq.wcm.core.components.models 12.8.0
+	 */
+	default String getModalId() {

```

Comment:
We discussed within the team and we think we should have a common approach for all components regarding IDs. We think it's best to have an author/defined ID. Could you please create a `CoreComponent` interface that exposes this via a `getId` method? Then we can add support for this in other components too, as a later step.

Reply:
Also see: https://github.com/adobe/aem-core-wcm-components/issues/535#issuecomment-483297494